### ISSUE TITLE
No More payment integrations! Everything is Gone! Its Trash!

### ISSUE THREAD

**Issue by lopezacuba** [akeeba/akeebasubs#389]
Why drop everything that made Akeeba Subscriptions so popular. Now its just trash! All the good components/plugins gone! Now Paddle is making money from Akeeba Subscriptions. Is the owner (Nicholas K. Dionysopoulos) of Akeeba now getting kickbacks (money) from putting Paddle in his program and removing ALL the other payment integrations? Mr. Dionysopoulos, please put everything back in Akeeba Subscriptions as it was before or it will become trash down-the-road. Nobody will use it and maybe not support Akeeba as much as they did before. The old saying, "If its not broke, Don't fix it!"

**Comment by Skullbock** [comment]
@lopezacuba Nicholas can do whatever he pleases with his software.

Also, as stated in the readme:

**"Internal project - No support"**

if you really want the old plugins, just use the previous version, or fork it and mantain it your self. It's FOSS so :)

Cheers

**Comment by nikosdion** [comment]
First of all, as I am sure you are fully aware, we [very publicly discontinued support and feature requests Akeeba Subscriptions in November 2013](https://www.akeebabackup.com/news/1534-the-future-of-akeebasubs.html). This was six years and 9 days before you filed this issue. Just on the basis of this announcement from six years ago your issue here is uncalled for.

Furthermore, as I have **CLEARLY AND EXPLICITLY** written in the project README since [March 23rd 2016](https://github.com/akeeba/akeebasubs/commit/4ed5ce22d21d4cff694695e88054e2139448cdd3#diff-04c6e90faac2675aa89e2176d2eec7d8) Akeeba Subscriptions is an internal project. The actual wording at the time read precisely the following (check the Git history, helpfully posted as a link above, if you do not believe me -- the commits cannot be rewritten, that's the whole point of Git):

> This is meant to be an internal software development project for use with our site, akeebabackup.com. As such, future versions of this software will drop any features we do not wish to maintain because we do not intend or anticipate to use on our site.

So, based on the clear admonishment from three years, six months and 19 days ago you have no reason to complain. You were given more than three years advance warning that Akeeba Subscriptions would radically change and features would be dropped.

There came the time that this had to happen. On [July 9th, 2019](https://github.com/akeeba/akeebasubs/commit/c6734c414bf3d9b75f9f71c3c218fe224dab969c#diff-04c6e90faac2675aa89e2176d2eec7d8) I did have to refactor the software in a radical way (more on that later) and as a result remove all the features we could no longer use. I do understand that this would be disruptive to anyone who, despite our stern warnings, still used Akeeba Subscriptions so I wrote the following very clear, straightforward and upfront admonishment in bold type at the top of the README:

> This version of Akeeba Subscriptions is dramatically different to versions 5 and 6. It is no longer a generic subscriptions component, it is a very tight integration with Paddle with features specific to our use case. Most features we did not use ourselves have been dropped. Please do not use unless you understand the implications.

So even if you missed the clear warnings from six and three years ago this should be more than enough to let you know that Akeeba Subscriptions 7 is most definitely not for you and you should not upgrade to it.

On the essence of the issue, no, we are OF COURSE not getting kickbacks from Paddle or any other payments processor. I have never IN MY LIFE accepted affiliate links or develop software that is either adware or forces people to use specific services. Shame on you for making this unfounded allegation. My entire life I've been giving people choices and freedom. When something had to be removed for any reason whatsoever I have always given people an advance warning and the reasoning behind it.

To put it simply, we had to use Paddle (or any other Merchant of Record reseller company) for legal reasons. In very simple terms, if we didn't do that I could actually go to jail. This is not an exaggeration.

A lot of countries now [require that we collect sales tax / VAT](https://paddle.com/support/which-countries-does-paddle-charge-vat-for/) for all software purchases made by their citizens or companies registered in them. This is despite us not having a physical or financial nexus in those countries. In plain English, we need to collect the taxes even though we are not registered in these countries and we do not have any financial ties with them. These taxes need to be paid back to these countries' (or state's, for USA) tax authorities. In practical terms, it would require us to register for tax in more than 60 jurisdictions _on top of European Union_. 

I did, in fact, ask our accountants if we could do that. Sure we could. The problem is that the cost for sales tax registration in each jurisdiction, filing sales taxes and the necessary upkeep was in the many dozens of thousands of Euros **per year**. This is more than the meager income we make from selling subscriptions. 

Since the income we have in some of these jurisdictions (USA, Australia and Switzerland for certain) is above the threshold where sales tax MUST be collected it put me in an impossible position. If I chose to not collect sales tax and continue with the previous version of Akeeba Subscriptions I would be at best fined at worst thrown to jail if I visited any of these countries -- countries that I do visit since we sponsor events there.

So here were our options as a company:

1. Face fines or jail time which would caused us to close up shop, ruin me financially and result in my loss of freedom.
1. Close up shop.
1. Go with a Merchant of Record company which handles billing, sales taxes and invoicing.

The first option is a no-go. Ruining my life because some random person on the Internet might be upset that I refactored our clearly marked internal use only project is not a possibility anyone would plausibly entertain.

The second option would be the end of all Akeeba software, free or paid. We would stop sponsoring events. We would stop contributing to Joomla. We'd no longer be in business. I would survive financially by doing something else. The bulk of my users would lose the software they depend on to protect their sites. This is a solution that sucks for everybody.

This left me with exactly one viable option, the third one: using a reseller (Merchant of Record) to remain in business and be tax compliant. I did consider various options between May 2018 and February 2019. That's how long I was looking into this. Charges levied per transaction ranged from approximately 7% all the way to 34%. Compared to 2.9% with Viva (cheapest option) and 5.4% with 2Checkout (most expensive option) we were talking about 50% to 100% more cost per transaction. Paddle was the cheapest of MoR companies. So no, not only I did not make more money by switching to Paddle, I actually lost money. It's just that the alternatives sucked even more. We lose more money but we can still make our software and sponsor events, even though this means moving to a smaller apartment and getting rid of our separate office. That is to say, this didn't just have a financial impact on our business, it also took a personal toll.

Given a sole option I had to figure out how to implement it. Any MoR company we'd choose would be handling billing, taxation and invoicing. I looked into whether it could be implemented as a plugin within the existing architecture of Akeeba Subs 6. It was not, in fact, possible. I had to refactor Akeeba Subscriptions and remove taxation, invoicing and do a tight integration around the MoR company we'd choose. This decision was made BEFORE going with Paddle and would still be true if we used any other MoR company such as DigitalRiver.

I was not happy that I had to both spend a lot of time working on our sales system instead of working on our products. This is the second year in a row I had to suffer such a diversion, the previous time being in 2018 when I had to come up with a GDPR compliance solution that Joomla completely lacked at the point. This meant that I was working long hours, weekends and holidays, spending far less time with my toddler than I would have liked to. So, once again, there is a very large element of personal cost. There's not enough money in the world to make up for the time I could not spend with my daughter. This makes your unfounded allegations all the more insulting.

I also did think about people who, despite our clear warnings from at least three and at most six years ago, were still using Akeeba Subscriptions. I honestly believe that a three year warning that Akeeba Subscriptions is an internal project is much more than adequate. If after three years you have not migrated away from it it's on you, not me. Still, Akeeba Subs 6 is still available for download and will continue to be maintained throughout the life of Joomla 3. It will receive no bug fixes and new features, just security updates. This gives you another 2 years to migrate away from it. Considering that the first warning to migrate away from it was given 6 years ago you get a total of 8 years to find a different solution. If that's not enough for you, sorry, my professionalism does not let me adequately express my opinion of you and your practices.

Finally, I'd like to address the tone of language and content of your issue. You have expressed yourself in a very unprofessional manner. Worse, you seem to be under the false impression that FOSS developers are somehow obligated to bend over to your whims and needs. I am afraid this is not the case. Moreover, your failure to take heed of my very clear warnings over the last six years is not something that reflects badly on me. If anything, the way you expressed yourself reflects very negatively on you.
